File: Projeto02/AtividadeSLN/Atividade/Transacao.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Projeto02
{
    internal class Transacao
    {
        public DateTime Data { get; set; }

        public string Tipo { get; set; }

        private decimal _valor;

        public decimal Valor
        {
            get { return _valor; }
            set
            {
                if (value > 0)
                    _valor = value;
                else
                    Console.WriteLine("Erro: O valor da transação deve ser positivo.");
            }
        }
    }
}

File: Projeto02/AtividadeSLN/Atividade/Transacao.py
from datetime import datetime

class Transacao:
    def __init__(self):
        self._valor = 0.0
        self.data = datetime.now()
        self.tipo = ""

    @property
    def valor(self):
        return self._valor

    @valor.setter
    def valor(self, value):
        # O valor deve ser obrigatoriamente maior que zero
        if value > 0:
            self._valor = value
        else:
            print("Erro: O valor da transaçăo deve ser positivo.")